File: activity/20200805/RabbitLee/Move_Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        quick, slow = 0, 0
        while quick < len(nums):
            if nums[quick] != 0:
                nums[slow] = nums[quick]
                slow += 1
            quick += 1
        while slow < len(nums):
            nums[slow] = 0
            slow += 1

File: activity/20200805/scau-flyingbird/MoveZeroes.go
func moveZeroes(nums []int) {
	idx := 0
	for i, num := range nums {
		if num != 0 {
			nums[idx] = nums[i]
			idx++
		}
	}
	for i := idx; i < len(nums); i++ {
		nums[i] = 0
	}
}
